docs/Repartitioner-sparkcompute.md

```diff
@@ -1,12 +1,54 @@
 # Repartitioner
 
-This plugins re-partitions a Spark RDD.
+This plugin repartitions incoming data. This changes the level of parallelism of the pipeline.
+
+Increasing partitions is usually done to increase parallelism when trying to speed up a pipeline.
+For example, the database source is often configured to use a relatively small number of partitions
+(tens, but not hundreds or thousands). This is to prevent a flood of concurrent connections to the database,
+which would overwhelm the database. However, once all the data is read, it can sometimes make sense to
+repartition to a much higher number to allow the rest of the pipeline to run with more parallelism.
+
+Decreasing partitions is done when you want to limit the amount of parallelism.
+For example, if you want all of the output written to a single file, you can decrease the partitions to 1
+before writing to a file sink. As another example, you may want to decrease the partitions when writing to a
+database sink to limit the amount of concurrent connections open to the database. As another example,
+you may want to decrease the partitions before using a transform that makes API calls to an external service,
+in order to avoid hitting rate limits or causing high load on that external service.
 
 ## Plugin Properties
 
-## Usage Notes
-Provide a clear description of your example Spark compute plugin.
+**Partitions:** Number of partitions to generate
```

Comment:
should we add default value in the docs?

Reply:
I haven't been documenting it when the default is set in the widget, in case it ever gets out of sync.

Reply:
these are also required settings right now, so there is no backend defaulting if it's not provided

Reply:
sounds good.